src/auth/src/credentials.rs

```diff
@@ -215,22 +215,17 @@ pub mod traits {
 /// [gcloud auth application-default]: https://cloud.google.com/sdk/gcloud/reference/auth/application-default
 /// [gke-link]: https://cloud.google.com/kubernetes-engine
 pub async fn create_access_token_credential() -> Result<Credential> {
-    let adc_path = adc_path().ok_or_else(||
-        // TODO(#442) - This should (successfully) fall back to MDS Credentials. We will temporarily return an error.
-        CredentialError::new(false, Box::from("Unable to find Application Default Credentials.")))?;
-
-    let contents = std::fs::read_to_string(adc_path).map_err(|e| {
-        match e.kind() {
-            std::io::ErrorKind::NotFound => {
-                // TODO(#442) - This should (successfully) fall back to MDS Credentials. We will temporarily return an error.
-                CredentialError::new(
-                    false,
-                    Box::from("Unable to find Application Default Credentials."),
-                )
-            }
-            _ => CredentialError::new(false, e.into()),
+    let adc_path = match adc_path() {
+        Some(path) => path,
+        None => {
+            return Ok(mds_credential::new());
         }
-    })?;
+    };
+    let contents = match std::fs::read_to_string(adc_path) {
+        Ok(contents) => contents,
+        Err(e) if e.kind() == std::io::ErrorKind::NotFound => return Ok(mds_credential::new()),
```

Comment:
So if `GOOGLE_APPLICATION_CREDENTIALS` points to a file that does not exist we fall back on MDS?  I think this:

https://google.aip.dev/auth/4110#expected-behavior

Says you should return an error.  But I find that diagram and the text very confusing.

Reply:
You are right. I did not read the spec closely.

I considered doing away with `fn adc_path()`, but instead decided to have it also return whether or not we are using the well known path.

I don't have a unit test for falling back to MDS if the well known path is not found. It seems like we don't have too much control over that. I could write a test that checks if the well known path is a file, and sees if we fallback based on that. Do you think that test is worth having?

https://doc.rust-lang.org/std/path/struct.Path.html#method.is_file